### PR TITLE
Issue 9570 create drop button component

### PR DESCRIPTION
**Pull request by freddyucv**

### CODE REVIEW

File: src/view/components/common/dropdown-component/dropdown-component.html [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<button md-icon-button (click)="onToggle()" [disabled]="disabled" *ngIf="icon">
```
Comment (fmontes) [inline src/view/components/common/dropdown-component/dropdown-component.html]: Here we are missing the title and the  ```<article>``` to wrap the components, please check the others.

Comment (fmontes) [inline src/view/components/common/dropdown-component/dropdown-component.html]: We are duplicating code here, you can do the ```*ngIf``` icon or title inside the ```<button>

File: src/view/components/common/dropdown-component/dropdown-component.scss [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+:host {
+  position: relative;
+
+  .dropdown-content {
+    @include box_shadow(2);
+    background-color: $white;
+    color: $black;
+    padding: 5px 10px 5px 10px;
+    z-index: 100;
+
```
Comment (fmontes) [inline src/view/components/common/dropdown-component/dropdown-component.scss]: Extra lines, alphabetically order for the properties

File: src/view/components/common/dropdown-component/dropdown-component.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
+import {Component, EventEmitter, Input, Output, ViewEncapsulation, ElementRef} from '@angular/core';
+
```
Comment (fmontes) [inline src/view/components/common/dropdown-component/dropdown-component.ts]: Extra lines.

File: src/view/components/common/dropdown-component/dropdown-component.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    providers: [],
+    selector: 'dot-dropdown-component',
+    styleUrls: ['dropdown-component.css'],
+    templateUrl: ['dropdown-component.html'],
+    host: {
+        '(document:click)': 'handleClick($event)',
+    }
+})
+
+/**
```
Comment (fmontes) [inline src/view/components/common/dropdown-component/dropdown-component.ts]: Remove unnecessary commented code.

File: src/view/components/common/pattern-library/pattern-library.html [hunk 5]
```diff
@@ -92,5 +92,86 @@ <h3 class="section__title">Radio Buttons</h3>
             </article>
         </div>
     </section>
+
+    <section class="row">
```
Comment (jdotcms) [inline src/view/components/common/pattern-library/pattern-library.html]: Shouldn't we i18n the messages?

File: src/view/components/common/dropdown-component/dropdown-component.ts [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    private show:boolean = false;
+
+    constructor (private elementRef: ElementRef){}
+
+    private onToggle():void{
+        this.show = !this.show;
+
+        if (this.show){
+            this.open.emit();
+        }else{
```
Comment (fmontes) [inline src/view/components/common/dropdown-component/dropdown-component.ts]: ```} else {```

File: src/view/components/common/dropdown-component/dropdown-component.ts [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+            this.close.emit();
+        }
+
+        this.toggle.emit( this.show );
```
Comment (fmontes) [inline src/view/components/common/dropdown-component/dropdown-component.ts]: ```(this.show)```

File: src/view/components/common/pattern-library/pattern-library.ts [hunk 8]
```diff
@@ -21,4 +22,12 @@ export class PatternLibrary {
     constructor() {
         console.log('PL')
```
Comment (jdotcms) [inline src/view/components/common/pattern-library/pattern-library.ts]: Usually Gamba prefer to remove the console.log's

File: src/view/components/common/dropdown-component/dropdown-component.ts [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+    }
+
+    private handleClick($event) {
```
Comment (fmontes) [inline src/view/components/common/dropdown-component/dropdown-component.ts]: Add a TODO here, we need to se how can we do this globally for all the components that need to detect if the click was outside it.

File: src/view/components/common/pattern-library/pattern-library.ts [hunk 10]
```diff
@@ -21,4 +22,12 @@ export class PatternLibrary {
...
     }
+
+    public print(message:string, argument:any){
```
Comment (fmontes) [inline src/view/components/common/pattern-library/pattern-library.ts]: Not sure what is this for.

**Pull request merged**